js_modules/dagit/packages/core/src/ui/SectionedLeftNav.tsx

```diff
@@ -0,0 +1,267 @@
+import {BaseTag, Box, Colors, Icon, IconWrapper, StyledTag} from '@dagster-io/ui';
+import * as React from 'react';
+import styled from 'styled-components/macro';
+
+import {AppContext} from '../app/AppContext';
+import {isAssetGroup} from '../asset-graph/Utils';
+import {useStateWithStorage} from '../hooks/useStateWithStorage';
+import {getJobItemsForOption, JobItem} from '../nav/FlatContentList';
+import {DagsterRepoOption, WorkspaceContext} from '../workspace/WorkspaceContext';
+import {buildRepoAddress} from '../workspace/buildRepoAddress';
+import {repoAddressAsString} from '../workspace/repoAddressAsString';
+import {RepoAddress} from '../workspace/types';
+
+const validateExpandedKeys = (parsed: unknown) => (Array.isArray(parsed) ? parsed : []);
+const EXPANDED_REPO_KEYS = 'dagit.expanded-repo-keys';
+
+export const SectionedLeftNav = () => {
+  const {loading, visibleRepos} = React.useContext(WorkspaceContext);
```

Comment:
I think it may be beyond the scope of this PR, but long term I'm not sure we want to have both `hidden` and `collapsed`, it may be ok to show all repos as collapsed all the time? The only scenario I can think of where you'd still want to HIDE a few is if you had a ton of them. (15+?)

Reply:
I think that many-repo case might be a genuine scenario that we need to cover. Either way, it's easy enough to change. :)